src/schoology-extractor/schoology_extractor/api/request_client.py

```diff
@@ -190,8 +194,12 @@ def get_users(self, page_size: int = 20) -> PaginatedResult:
             A parsed response from the server
 
         """
-        url = f"users?start=0&limit={page_size}"
+        url = f"users?{self._build_pagination_params(page_size)}"
 
         return PaginatedResult(
             self, page_size, self.get(url), "user", self.base_url + url
         )
+
+    def _build_pagination_params(self, page_size: int):
+        assert isinstance(page_size, int)
+        return f"start=0&limit={page_size}"
```

Comment:
How do you get the second page, if the `start` is always zero?

Reply:
This parameter is only called the first time, for the next requests it will get the next page link from the Schoology's response

Reply:
Ok, that makes good sense now. Suggestion: update the function name so the reader will understand this clearly. Perhaps `_build_query_params_for_first_page`?

FYI, generally I do not write argument validations for "private" methods. Why? It is a compromise: instead of saying that _all_ functions must validate _all_ arguments, I usually ask that _public_ (and _protected_ in C#/Java/etc) functions need argument validation. With private functions, and a small class, it is usually easy to see that the values are already validated elsewhere.

So, there's no reason to delete the validation right now. Just letting you know that I would not have complained if it had not been there.

Reply:
Renamed ✅